### ISSUE TITLE
The round_robin loadbalancer is not working if DNS TXT record resolves failed with an serviceConfigError

### ISSUE THREAD

**Issue by amit-gshe** [grpc/grpc-node#1510]
When I use round_robin load balancer, if DNS TXT record resolves failed with an serviceConfigError then the service config I configured was override by the default pick_first policy config.
My grpc channel config is the following:
```
{
'grpc.service_config': JSON.stringify({
    loadBalancingPolicy: 'round_robin',
    loadBalancingConfig: [{ name: 'round_robin', round_robin: {} }],
  })
}
```
From the source:
https://github.com/grpc/grpc-node/blob/301b695d1552918629d50ee434a9e3269fac607d/packages/grpc-js/src/resolving-load-balancer.ts#L119
The first time the `serviceConfig` and `serviceConfigError` is null and the `this.previousServiceConfig` was set to null. The second time the `serviceConfigError` is not null because of failling TXT records resolving then the working service config was replaced by the `this.previousServiceConfig` which is null now so working config was fallback to pick_first config.
https://github.com/grpc/grpc-node/blob/301b695d1552918629d50ee434a9e3269fac607d/packages/grpc-js/src/resolving-load-balancer.ts#L140
should be replaced by `if (!this.defaultServiceConfig)` here IMO.

**Comment by murgatroid99** [comment]
I see what happened here. All of the DNS responses used to be handled atomically in a single `onSuccessfulResolution`, but that was changed in #1316 to address a situation where DNS servers were taking a very long time to respond negatively to TXT records and that was delaying connection establishment for no benefit. I missed how that would impact the default service config.

Interestingly, the [document](https://github.com/grpc/proposal/blob/master/A21-service-config-error-handling.md) referenced in the section that you linked has actually had a couple of revisions so the behavior of that section needs to change a little. And based on the current version of that document, I think the real bug is actually here:

https://github.com/grpc/grpc-node/blob/301b695d1552918629d50ee434a9e3269fac607d/packages/grpc-js/src/resolver-dns.ts#L239-L244

It looks like the `serviceConfigError` should be used exclusively for validation errors, and failing to retrieve the TXT records should be handled differently. This is especially true because, if I remember correctly, Node treats the absence of a TXT record as a failure.

**Comment by murgatroid99** [comment]
I have published version 1.1.3 with a change that I think will fix this.
